pkg/network/dns/stats.go

```diff
@@ -168,20 +170,14 @@ func (d *dnsStatKeeper) ProcessPacketInfo(info dnsPacketInfo, ts time.Time) {
 	d.stats[info.key] = allStats
 }
 
-func (d *dnsStatKeeper) GetNumStats() (int32, int32) {
-	numStats := d.lastNumStats.Load()
-	droppedStats := d.lastDroppedStats.Load()
-	return numStats, droppedStats
-}
-
 func (d *dnsStatKeeper) GetAndResetAllStats() StatsByKeyByNameByType {
 	d.mux.Lock()
 	defer d.mux.Unlock()
 	ret := d.stats // No deep copy needed since `d.stats` gets reset
 	d.stats = make(StatsByKeyByNameByType)
 	log.Debugf("[DNS Stats] Number of processed stats: %d, Number of dropped stats: %d", d.numStats, d.droppedStats)
-	d.lastNumStats.Store(int32(d.numStats))
-	d.lastDroppedStats.Store(int32(d.droppedStats))
+	statsTelemetry.lastNumStats.Set(float64(d.numStats))
+	statsTelemetry.lastDroppedStats.Set(float64(d.droppedStats))
```

Comment:
I think we are going to see weird behavior with this metric, because prometheus is not synchronized to calls to `GetAndResetAllStats`. I think the only reason it is coded this way, is to support the log message.

I think if you inverted the relationship, so you are incrementing the telemetry values rather than `numStats` and `droppedStats`, and then computed the deltas when this is called for outputting in the log message, it would work better.

Reply:
Let me know if that isn't clear and I can write some example code to show what I mean.

Reply:
fixed (I think)